packages/aws-cdk-lib/aws-events-targets/lib/redshift-query.ts

```diff
@@ -0,0 +1,138 @@
+import { bindBaseTargetConfig, singletonEventRole } from './util';
+import * as events from '../../aws-events';
+import * as iam from '../../aws-iam';
+import * as secretsmanager from '../../aws-secretsmanager';
+import * as sqs from '../../aws-sqs';
+
+/**
+ * Configuration properties of an Amazon Redshift Query event.
+ */
+export interface RedshiftQueryProps {
+
+  /**
+   * The Amazon Redshift database to run the query against.
+   */
+  readonly database: string;
+
+  /**
+   * The Amazon Redshift database user to run the query as. This is required when authenticating via temporary credentials.
+   *
+   * @default - No Database user is specified
+   */
+  readonly dbUser?: string;
+
+  /**
+   * The secret containing the password for the database user. This is required when authenticating via Secrets Manager.
+   * If the full secret ARN is not specified, this will instead use the secret name.
+   *
+   * @default - No secret is specified
+   */
+  readonly secret?: secretsmanager.ISecret;
+
+  /**
+   * The SQL query to run. This will use the `executeStatement` API.
+   *
+   * @default - No SQL query is specified
+   */
+  readonly sql?: string;
```

Comment:
They are separate in the EventBridge [API](https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_RedshiftDataParameters.html) there are use cases when a user might want to run multiple statements as a single transaction (think ACID) vs each statement as a separate transaction, which is why I am guessing there's two different attributes on the API.

Reply:
Pretty much what @manmartgarc mentioned. It's for ACID compliant transactions. I could change the interface to instead take in a single `sql` array that, when multiple string elements are parsed, are sent via the `batchSql` API. If a single element is parsed, then we use the `sql` API. What do you think @paulhcsun

Reply:
Gotcha, ya I would prefer to just expose a single `sql` array property. I can check with someone on the Eventbridge service team to see if there's a need need to use the single `sql` API or if we can just pass everything through the `batchSQL` query if it does the same thing. It's likely that behind the scenes the `batchSQL` API is just calling the `sql` API.

Reply:
there's 100% use case for a user to want to treat many SQL statements as either separate or collective transactions. Indeed an array of length 1 passed to `batchSQL` should be equivalent to using the `sql` parameter; even if the passed string has many statements together.

Reply:
The engineer on the Eventbridge team said that they're just calling either the Redshift `BatchExecuteStatement` or `ExecuteStatement` APIs for `Sqls` and `Sql` respectively so I'm waiting on a response from the Redshift service team now. 

@manmartgarc Sorry I may be reading too much into it but is that to imply you think we should still have a separate `sql` and `batchSQL` parameter or that since they're equivalent it should be fine to just have a single `sql` parameter in CDK?

Reply:
Yes that's what I thought, although it looks like I am wrong. The [Redshift Data API](https://docs.aws.amazon.com/redshift-data/latest/APIReference/API_ExecuteStatement.html) states for `ExecuteStatement`

> This statement must be a single SQL statement.

So the case where I say that an array of length 1 with _many_ statements via `BatchExecuteStatement` is equivalent to a single call to `ExecuteStatement` is not correct since each statement, either one through `ExecuteStatement` or many (one for each element in the array) through `BatchExecuteStatement` must actually be a _single_ statement.

A statement is:

```SQL
SELECT *
FROM my table;
```

where many in a single string (what I thought could be done) is:

```SQL
SELECT *
FROM A;
SELECT *
FROM B;
```

It looks like indeed there's a use case to support both `sql` and `batchSQL` arguments.

Reply:
given what you said @manmartgarc, we could still include in the interface a single `sql` string array. Then if the length == 1, we could set the `sql` prop, otherwise, set the `batchSql` prop (will probably turn `batchSQL` to `batchSql`).

Reply:
@paulhcsun, what do you think of the above suggestion?

Reply:
> given what you said @manmartgarc, we could still include in the interface a single sql string array. Then if the length == 1, we could set the sql prop, otherwise, set the sqls prop

Good point @manmartgarc, let's go with the suggested solution then!

Reply:
I've implemented this now